common/src/main/java/xyz/wagyourtail/jsmacros/client/api/library/impl/FWorld.java

```diff
@@ -351,6 +533,69 @@ public long getTimeOfDay() {
         assert mc.world != null;
         return mc.world.getTimeOfDay();
     }
+
+    /**
+     * @return {@code true} if it is daytime, {@code false} otherwise.
+     *
+     * @since 1.8.4
+     */
+    public boolean isDay() {
+        assert mc.world != null;
+        return mc.world.isDay();
+    }
+
+    /**
+     * @return {@code true} if it is nighttime, {@code false} otherwise.
+     *
+     * @since 1.8.4
+     */
+    public boolean isNight() {
+        assert mc.world != null;
+        return mc.world.isNight();
+    }
+
+    /**
+     * @return {@code true} if it is raining, {@code false} otherwise.
+     *
+     * @since 1.8.4
+     */
+    public boolean isRaining() {
+        assert mc.world != null;
+        return mc.world.isRaining();
+    }
+
+    /**
+     * @return {@code true} if it is thundering, {@code false} otherwise.
+     *
+     * @since 1.8.4
+     */
+    public boolean isThundering() {
+        assert mc.world != null;
+        return mc.world.isThundering();
+    }
+
+    /**
+     * @return the name of the loaded world or {@code "UNKNOWN_NAME"} if no name could be found.
+     *
+     * @since 1.8.4
+     */
+    public String getServerName() {
+        IntegratedServer server = mc.getServer();
+        if (server != null) {
+            return "LOCAL_" + server.getSavePath(WorldSavePath.ROOT).normalize().getFileName();
+        }
+        ServerInfo multiplayerServer = mc.getCurrentServerEntry();
+        if (multiplayerServer != null) {
+            if (mc.isConnectedToRealms()) {
+                return "REALM_" + multiplayerServer.name;
+            }
+            if (multiplayerServer.isLocal()) {
+                return "LAN_" + multiplayerServer.name;
+            }
+            return multiplayerServer.address.replace(":25565", "").replace(":", "_");
+        }
+        return "UNKNOWN_NAME";
```

Comment:
did you grab this from 
https://github.com/wagyourtail/WagYourMinimap/blob/7a882e366cd588bc90110b63e0fbe3bb1ee27741/common/src/main/java/xyz/wagyourtail/minimap/api/client/MinimapClientApi.java#L30

and is that a good idea? this isn't the same as the file name for singleplayer worlds

Reply:
Yes, I have taken the code from your Minimap mod.
I have renamed the function to `getWorldIdentifier` with the note that the identifier is mostly unique. Thinking about this, it would be more robust to also use the address for the realm and lan servers.

Reply:
ip address for realms is dynamic, which is why I did this in the first place, had some issues in some other mods.
same for lan

Reply:
Good to know, makes sense that Realms have dynamic IPs. I also updated the Javadoc to make it more clear that this method should produce an identifier that is probably unique enough to identify worlds, but not guarenteed.